correlation/src/main/java/com/cerner/beadledom/correlation/CorrelationIdContext.java

```diff
@@ -0,0 +1,68 @@
+package com.cerner.beadledom.correlation;
+
+import java.io.Closeable;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * A context that provides access to the Correlation ID.
+ */
+public class CorrelationIdContext {
+
+  private static final ThreadLocal<String> correlationId = new ThreadLocal<>();
+
+  private CorrelationIdContext() {
+  }
+
+  /**
+   * Returns the current Correlation ID, {@code null} if no Correlation ID is in context.
+   */
+  @Nullable
+  public static String get() {
```

Comment:
I was surprised to see all these methods are static. In other cases where we have used `ThreadLocal` we have encapsulated it into instances. Is there a reason for not having usages of this class operate on an instance?

Reply:
I originally did use instances, https://github.com/cerner/beadledom/blob/520830961d11c6e79c18896eee02f5f45f557dc7/correlation/src/main/java/com/cerner/beadledom/correlation/ThreadLocalCorrelationContext.java and ended up switching to static methods for simplicity for the consumer. Consumers may want to add additional functionality to the implementation, which would be afforded if it was a instance. I'll work on switching it back.

Reply:
Updated CorrelationIdContext to be a interface and created a implementation. Is that what you were thinking?

https://github.com/cerner/beadledom/pull/180/commits/8a7b63eb74bda01165675f0a39c69f30d5ab6f9d